user/src/com/google/gwt/core/client/ScriptInjector.java

```diff
@@ -265,6 +272,7 @@ function clearCallbacks() {
       }
     }
     scriptElement.onload = $entry(function() {
+      scriptRegistry[scriptElement.src] = scriptElement;
```

Comment:
Pretty sure you would want to set this before it finishes loading, otherwise there will be a race condition if you add the same script again before the first one finishes loading.